codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -230,86 +263,107 @@ public static ByteBuf decode(
             throw new NullPointerException("dialect");
         }
 
-        byte[] DECODABET = decodabet(dialect);
+        // Using a ByteProcessor to reduce bound and reference count checking.
+        return new Decoder().decode(src, off, len, allocator, dialect);
+    }
 
-        int len34 = len * 3 / 4;
-        ByteBuf dest = allocator.buffer(len34).order(src.order()); // Upper limit on size of output
-        int outBuffPosn = 0;
+    private static final class Decoder implements ByteProcessor {
 
-        byte[] b4 = new byte[4];
-        int b4Posn = 0;
-        int i;
-        byte sbiCrop;
-        byte sbiDecode;
-        for (i = off; i < off + len; i ++) {
-            sbiCrop = (byte) (src.getByte(i) & 0x7f); // Only the low seven bits
-            sbiDecode = DECODABET[sbiCrop];
+        private final byte[] b4 = new byte[4];
+        private int b4Posn;
+        private byte sbiCrop;
+        private byte sbiDecode;
+        private byte[] decodabet;
+        private int outBuffPosn;
+        private ByteBuf dest;
+
+        ByteBuf decode(ByteBuf src, int off, int len, ByteBufAllocator allocator, Base64Dialect dialect) {
+            int len34 = (len * 3) >>> 2;
+            dest = allocator.buffer(len34).order(src.order()); // Upper limit on size of output
```

Comment:
oh it was there when I looked ... and looked like https://github.com/netty/netty/pull/6404#discussion_r101837914 ... good catch @johnou